### ISSUE TITLE
missing_transmute_annotations wrong order of Src and Dst

### ISSUE THREAD

**Issue by slonkazoid** [rust-lang/rust-clippy#13215]
### Summary

the [missing_transmute_annotations](https://github.com/rust-lang/rust-clippy/blob/cc63143bbf5ae28daae127902090492560313ca1/clippy_lints/src/transmute/missing_transmute_annotations.rs) suggests the wrong order of Src and Dst generics for `transmute::<Src, Dst>()` when i try to cast a u8 array into a tuple of integers

### Reproducer

I tried this code:

```rust
pub fn fixup_uuid(data: u128) -> u128 {
    let (first, second, third, fourth): (u32, u16, u16, u64) =
        unsafe { std::mem::transmute(data.to_be_bytes()) };
    let real = unsafe {
        std::mem::transmute((
            first.swap_bytes(),
            second.swap_bytes(),
            third.swap_bytes(),
            fourth,
        ))
    };
    u128::from_be_bytes(real)
}
```

I expected to see this happen:

it should have told me to write the transmute like this:  
`transmute::<[u8; 16], (u32, u16, u16, u64)>`

Instead, this happened:

```rust
warning: transmute used without annotations
 --> src/util.rs:5:19
  |
5 |         std::mem::transmute((
  |                   ^^^^^^^^^ help: consider adding missing annotations: `transmute::<(u32, u16, u16, u64), [u8; 16]>`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_transmute_annotations
  = note: `#[warn(clippy::missing_transmute_annotations)]` on by default
```

### Version

```yaml
rustc 1.82.0-nightly (8e86c9567 2024-08-01)
binary: rustc
commit-hash: 8e86c9567154dc5a9ada15ab196d23eae2bd7d89
commit-date: 2024-08-01
host: x86_64-unknown-linux-gnu
release: 1.82.0-nightly
LLVM version: 19.1.0
```

**Comment by slonkazoid** [comment]
nevermind i missed the line number :)
